pkg/controller/complianceremediation/complianceremediation_controller.go

```diff
@@ -378,6 +395,123 @@ func (r *ReconcileComplianceRemediation) countKubeUnmetDependencies(rem *compv1a
+		labels[compv1alpha1.RemediationValueRequiredProcessedLabel] = ""
+		if len(unsetRequireValueList) > 0 {
+			annotations[compv1alpha1.RemediationUnsetValueAnnotation] = strings.Join(append(currentUnsetValues, unsetRequireValueList...), ",")
+		}
+		if len(setRequireValueList) > 0 {
+			annotations[compv1alpha1.RemediationValueUsedAnnotation] = strings.Join(append(currentUsedValues, setRequireValueList...), ",")
+		}
+		logger.Info("Labeling remediation to denote some values haven't been set")
+		rCopy := rem.DeepCopy()
+		rCopy.SetAnnotations(annotations)
+		rCopy.SetLabels(labels)
+		err := r.client.Update(context.TODO(), rCopy)
+		if err != nil {
+			return hasUpdate, reconcile.Result{}, fmt.Errorf("adding xccdf required-value label/annotation: %w", err)
+		}
+		hasUpdate = true
+		return hasUpdate, reconcile.Result{}, nil
+	}
+	return hasUpdate, reconcile.Result{}, nil
+}
+
+//To find if it is set in tailored profile
+func (r *ReconcileComplianceRemediation) isRequiredValueSet(rem *compv1alpha1.ComplianceRemediation, requiredValue string) (bool, error) {
+	var scan = &compv1alpha1.ComplianceScan{}
+	err := r.client.Get(context.TODO(), types.NamespacedName{
+		Namespace: rem.GetNamespace(),
+		Name:      rem.GetScan(),
+	}, scan)
+
+	if err != nil {
+		return false, err //error getting scan
+	}
+	if scan.Spec.TailoringConfigMap != nil {
+		tp := &compv1alpha1.TailoredProfile{}
+		err = r.client.Get(context.TODO(), types.NamespacedName{Name: scan.Spec.TailoringConfigMap.Name, Namespace: rem.GetNamespace()}, tp)
+		if err != nil {
+			return false, err
+		}
+		for _, setValue := range tp.Spec.SetValues {
+			if setValue.Name == requiredValue {
+				return true, nil
+			}
+		}
+		return false, nil
+	}
+	//scan did not have a tailored profile
+	return false, nil
+}
+
+func (r *ReconcileComplianceRemediation) handleUnsetValues(rem *compv1alpha1.ComplianceRemediation, logger logr.Logger) (bool, reconcile.Result, error) {
+	nNotSetValues := r.countXCCDFUnsetValues(rem, logger)
+	hasUpdate := false
+	labels := rem.GetLabels()
+	if labels == nil {
+		labels = make(map[string]string)
+	}
+	//To avoide setting incorrect label for the one that has required-value annotation but does not have any un-set values
+	if nNotSetValues > 0 {
+		//Set unset lable if labes has not been set
+		if _, ok := labels[compv1alpha1.RemediationUnsetValueLabel]; !ok {
+			labels[compv1alpha1.RemediationUnsetValueLabel] = ""
+			rCopy := rem.DeepCopy()
+			rCopy.SetLabels(labels)
+			err := r.client.Update(context.TODO(), rCopy)
+			if err != nil {
+				return hasUpdate, reconcile.Result{}, fmt.Errorf("adding un-set xccdf value label: %w", err)
+			}
+			hasUpdate = true
+			return hasUpdate, reconcile.Result{}, nil
+		}
+		return hasUpdate, reconcile.Result{}, nil
+	}
+	return hasUpdate, reconcile.Result{}, nil
+}
+
+func (r *ReconcileComplianceRemediation) countXCCDFUnsetValues(rem *compv1alpha1.ComplianceRemediation, logger logr.Logger) int {
```

Comment:
logger is unused in this function. I guess it also doesn't have to me a method of `r` but it's not harmful either and we haven't been super consistent with how we use functions and methods.

Reply:
I think we can remove that logger here since it is not used